### PR TITLE
Feat/admin invoices show v3

### LINKED ISSUE #35: Admin Invoice Show Page
Admin Invoice Show Page

As an admin,
When I visit an admin invoice show page
Then I see information related to that invoice including:
- Invoice id
- Invoice status
- Invoice created_at date in the format "Monday, July 18, 2019"
- Customer first and last name

### LINKED ISSUE #36: Admin Invoice Show Page: Invoice Item Information
Admin Invoice Show Page: Invoice Item Information

As an admin
When I visit an admin invoice show page
Then I see all of the items on the invoice including:
- Item name
- The quantity of the item ordered
- The price the Item sold for
- The Invoice Item status

### LINKED ISSUE #37: Admin Invoice Show Page: Total Revenue
Admin Invoice Show Page: Total Revenue

As an admin
When I visit an admin invoice show page
Then I see the total revenue that will be generated from this invoice

### PR DESCRIPTION
**Pull request by lcole37**
Admin invoice show page
create test and view for show page
*list invoice information
*list all invoice items and info
*display total revenue
*test total revenue model method
Tested and passed

### CODE REVIEW

File: app/views/admin/invoices/show.html.erb [hunk 1]
```diff
@@ -0,0 +1,17 @@
+<h1> Admin Dashboard </h1>
+
+<h2> Invoice ID: <%= @invoice.id %> </h2>
+<p> Status: <%= @invoice.status %> </p>
+<p> Created at: <%= @invoice.created_at.strftime("%A, %B %d, %Y") %> </p>
+<h3> Customer: <%= "#{@invoice.customer.first_name} #{@invoice.customer.last_name}" %> </h3>
```
Comment (Dominicod) [inline app/views/admin/invoices/show.html.erb:6]: To make make this a bit cleaner you could instantiate customer in the controller and then only have customer.first_name/last_name however this works!

File: app/views/admin/invoices/show.html.erb [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+<br>
+<h2> Invoice Items </h2>
+
+<% @invoice.invoice_items.each do |invoice_item| %>
+  <h3> Item: <%= invoice_item.item.name%> </h3>
+  <p> Quantity: <%= invoice_item.quantity %> </p>
+  <p> Unit Price: $<%= '%.2f'% ((invoice_item.item.unit_price.to_f) / 100)  %> </p>
```
Comment (Dominicod) [inline app/views/admin/invoices/show.html.erb:13]: Same thing in this section, possibly instantiating the invoice_items would clean this up, only concern is Law of Demeter, as we are reaching through three objects to get to item here.

File: spec/features/admin/invoices/show_spec.rb [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+      visit admin_invoice_path(Invoice.first)
+      # save_and_open_page
+      expect(page).to have_content("Invoice Items")
+      expect(page).to have_content("Item Qui Esse")
+      expect(page).to have_content("Quantity: 5")
+      expect(page).to have_content("Unit Price: $751.07")
+      expect(page).to have_content("Status: shipped")
+
+      expect(page).not_to have_content("Item Eos Quia")
```
Comment (Dominicod) [inline spec/features/admin/invoices/show_spec.rb:37]: Awesome job checking for .not_to !

File: spec/features/admin/invoices/show_spec.rb [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+  describe "When I visit an admin invoice show page" do
+    it "Then I see information related to that invoice including:
+      - Invoice id
+      - Invoice status
+      - Invoice created_at date in the format 'Monday, July 18, 2019'
+      - Customer first and last name" do
+
+      visit admin_invoice_path(Invoice.first)
+
+      expect(page).to have_content("Invoice ID: 1")
```
Comment (Dominicod) [inline spec/features/admin/invoices/show_spec.rb:13]: Is invoice ID suppose to be checked twice?

**Review by Dominicod — APPROVED** [review]
Looks good for me! Left a few comments that could help improve but I think its ready to merge.

File: spec/features/admin/invoices/show_spec.rb (hunk 4 above)
Comment (thayes87) [reply spec/features/admin/invoices/show_spec.rb:13]: agreed. Looks like lines 13  14 are duplicated.

File: app/views/admin/invoices/show.html.erb [hunk 5]
```diff
@@ -0,0 +1,17 @@
+<h1> Admin Dashboard </h1>
+
+<h2> Invoice ID: <%= @invoice.id %> </h2>
+<p> Status: <%= @invoice.status %> </p>
+<p> Created at: <%= @invoice.created_at.strftime("%A, %B %d, %Y") %> </p>
```
Comment (thayes87) [inline app/views/admin/invoices/show.html.erb:5]: nice work formatting the date with strftime("%A, %B %d, %Y")

**Review by thayes87 — APPROVED** [review]

File: spec/features/admin/invoices/show_spec.rb (hunk 4 above)
Comment (lcole37) [reply spec/features/admin/invoices/show_spec.rb:13]: good catch, thank you.

**Pull request merged**